### ISSUE TITLE
`docker start -a` does not start containers

### ISSUE THREAD

**Issue by markrechler** [docker/cli#2989]
**Description**

**Steps to reproduce the issue:**
1. `docker -D start -a logstash-collector`
2.
```
~# docker ps
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
```
3. `^CDEBU[0211] Error sending signal: Error response from daemon: Cannot kill container: bf24c7dfa6ada0bc90bae19309427c9cd16bbd1832f178a964461c318788665d: Container bf24c7dfa6ada0bc90bae19309427c9cd16bbd1832f178a964461c318788665d is not running`

**Describe the results you received:**
A docker start process attached to no running container.

**Describe the results you expected:**
A process attached to a running container. Works as expected in `19.03.15`.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
~# docker version
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        d3cb89e
 Built:             Thu Feb 25 07:05:21 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Thu Feb 25 07:03:17 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker info`:**

```
~# docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 2
 Server Version: 20.10.4
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-1038-aws
 Operating System: Ubuntu 18.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.786GiB
 Name: data-eng-interview-102
 ID: CUQK:L2L6:7DHX:A2V4:UAZK:Q55Z:SLGQ:2U2G:JHGZ:HFCD:H73O:GYQ6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running in AWS on Ubuntu Bionic 18.04

**Comment by fcastilloec** [comment]
I think this PR will fix this issue #2987

**Comment by thaJeztah** [comment]
Yes, this should be fixed in docker 20.10.5 (through the PR you linked); bug was in the CLI (not the daemon)
